Extensions/xkit_preferences.js

```diff
@@ -378,14 +400,10 @@ XKit.extensions.xkit_preferences = new Object({
 			var show_all = XKit.tools.get_setting("xkit_show_feature_updates","true") === "true";
 
 			var m_return = 0;
+
 			for (i=0;i<prev_objects.length;i++) {
```

Comment:
We're basically counting all the items that aren't read and aren't hidden, right? I wonder if we can do away with this entire `for` loop and just have

```js
var unread_items = prev_objects.filter(function(item){
    return !item.read && !XKit.extensions.xkit_preferences.news.hide_news_entry(item);
});
return unread_items.length;
```

I hope `XKit.extensions.xkit_preferences.news.hide_news_entry` is accessible inside the anonymous function though. Definitely want to double-check this first.

If this doesn't work, then don't worry about it.